test_pytest_html.py

```diff
@@ -25,22 +25,48 @@ def run(testdir, path='report.html', *args):
     return result, html
 
 
-def assert_summary(html, tests=1, duration=None, passed=1, skipped=0, failed=0,
+def assert_results_by_outcome(html, test_outcome, test_outcome_number,
+                              label=None):
+    # Asserts if the test number of this outcome in the summary matches
+    regex_summary = '(\d)+ {0}'.format(label or test_outcome)
+    int(re.search(regex_summary, html).group(1)) == test_outcome_number
+
+    # Asserts if the generated checkboxes of this outcome matches
+    regex_checkbox = ('<input checked="true" data-test-result="{0}"'
+                      .format(test_outcome))
+    if test_outcome_number == 0:
+        regex_checkbox += ' disabled="true"'
+    assert str(re.search(regex_checkbox, html).group()) == regex_checkbox
```

Comment:
No need to do a string compare here, you can just assert that `re.search ... is not None`